Controls/Pickers/ComboBox.coffee

```diff
@@ -0,0 +1,217 @@
+# ComboBox requires these modules. Please include them in your /modules directory
+{Type} = require "Type"
+{Color} = require "Color"
+
+comboBoxWidth = 296
+itemHeight = 32
+
+class suggestItem extends Layer
+	@define "item",
+		get: -> if @string? then @string.text
+		set: (value) -> if @string? then @string.text = value
+
+	@define "selected",
+		get: -> if @? then return @_selected
+		set: (value) -> if @? then @_selected = value
+
+	constructor: (options) ->
+		super _.defaults options,
+			backgroundColor: undefined
+			width: comboBoxWidth
+			height: itemHeight
+			item: "item"
+			_selected: false
+		@createItemLayer()
+		@item = options.item
+		@selected = options.selected
+
+	createItemLayer: ->
+		@backgroundColor = @setBackgroundColor("rest")
+
+		@string = new Type
+			parent: @
+			name: "item string"
+			x: 12
+			y: 5
+			uwpStyle: "body"
+			textOverflow: 'clip'
+			text: "something"
+
+		@onMouseOver -> @backgroundColor = @setBackgroundColor("over")
+		@onMouseDown -> @backgroundColor = @setBackgroundColor("down")
+		@onMouseUp -> @backgroundColor = @setBackgroundColor("over")
+		@onMouseOut -> @backgroundColor = @setBackgroundColor("rest")
+		@onTap -> @emit("itemTapped")
+
+	setBackgroundColor: (state) ->
+		switch state
+			when "rest"
+				if @selected then Color.listAccentLow else Color.transparent
+			when "over"
+				if @selected then Color.listAccentMedium else Color.listLow
+			when "down"
+				if @selected then Color.listAccentHigh else Color.listMedium
+
+class exports.ComboBox extends Layer
+	# A default list to get our customers going
+	# The list should contain 2 values: 'item' and 'selected'
```

Comment:
How about one comment:
`The default list contains 2 values: 'item' and 'selected'`